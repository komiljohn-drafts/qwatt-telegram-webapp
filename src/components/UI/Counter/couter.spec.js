import React from 'react'
import { render } from '@testing-library/react'
import { Counter } from './index'
import { mockStore } from '@/test-config/mockReduxStore'
import MockReduxProvider from '@/test-config/MockReduxProvider'

describe('Counter component', () => {
  it('should render Counter component with values', () => {
    const { asFragment } = render(
      <MockReduxProvider customStore={mockStore}>
        <Counter />
      </MockReduxProvider>
    )
    expect(
      asFragment(
        <MockReduxProvider customStore={mockStore}>
          <Counter />
        </MockReduxProvider>
      )
    ).toMatchSnapshot()
  })

  it('should render Counter component with empty values', () => {
    const { asFragment } = render(
      <MockReduxProvider>
        <Counter />
      </MockReduxProvider>
    )
    expect(
      asFragment(
        <MockReduxProvider>
          <Counter />
        </MockReduxProvider>
      )
    ).toMatchSnapshot()
  })
})
