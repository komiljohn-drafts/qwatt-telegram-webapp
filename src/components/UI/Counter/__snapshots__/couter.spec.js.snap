// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should render Counter component with empty values 1`] = `
<DocumentFragment>
  <div>
    <div
      class="row"
    >
      <button
        aria-label="Increment value"
        class="button"
      >
        +
      </button>
      <span
        class="value"
      >
        0
      </span>
      <button
        aria-label="Decrement value"
        class="button"
      >
        -
      </button>
    </div>
    <div
      class="row"
    >
      <input
        aria-label="Set increment amount"
        class="textbox"
        value="2"
      />
      <button
        class="button"
      >
        Add Amount
      </button>
      <button
        class="asyncButton"
      >
        Add Async
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Counter component should render Counter component with values 1`] = `
<DocumentFragment>
  <div>
    <div
      class="row"
    >
      <button
        aria-label="Increment value"
        class="button"
      >
        +
      </button>
      <span
        class="value"
      >
        4
      </span>
      <button
        aria-label="Decrement value"
        class="button"
      >
        -
      </button>
    </div>
    <div
      class="row"
    >
      <input
        aria-label="Set increment amount"
        class="textbox"
        value="2"
      />
      <button
        class="button"
      >
        Add Amount
      </button>
      <button
        class="asyncButton"
      >
        Add Async
      </button>
    </div>
  </div>
</DocumentFragment>
`;
